src/conn_handler.c

```diff
@@ -460,21 +462,23 @@ static int handle_ascii_client_connect(statsite_conn_handler *handle) {
             goto ERR_RET;
         }
 
-        // Handle counter sampling if applicable
-        if (type == COUNTER && !buffer_after_terminator(type_str, after_len, '@', &sample_str, &after_len)) {
+        // Handle counter/timer sampling if applicable
+        if ((type == COUNTER || type == TIMER) && !buffer_after_terminator(type_str, after_len, '@', &sample_str, &after_len)) {
             sample_rate = str2double(sample_str, &endptr);
-            if (unlikely(endptr == sample_str)) {
-                syslog(LOG_WARNING, "Failed sample rate conversion! Input: %s", sample_str);
-                goto ERR_RET;
-            }
-            if (sample_rate > 0 && sample_rate <= 1) {
+            if (type == COUNTER) {
```

Comment:
This should probably guard the value maximization, since we still want to check for the error in the TIMER case